assets/scripts/streets/StreetMetaData.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import { formatDate } from '../util/date_format'
+import { msg } from '../app/messages'
+import { getSignInData, isSignedIn } from '../users/authentication'
+import { getRemixOnFirstEdit } from './remix'
+import { showGallery } from '../gallery/view'
+import StreetWidth from './StreetWidth'
+import Avatar from '../app/Avatar'
+
+export default class StreetMetaData extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      street: this.props.street
+    }
+    this.onAuthorClick = this.onAuthorClick.bind(this)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.setState({
+      street: nextProps.street
+    })
+  }
+
+  onAuthorClick (e) {
```

Comment:
I prefer the naming convention `on${eventType}${context}` , e.g. `onClickAuthor()` instead of `onAuthorClick()`.